libraries/AP_Compass/AP_Compass_BMM150.cpp

```diff
@@ -314,47 +314,14 @@ void AP_Compass_BMM150::_update()
     /* convert uT to milligauss */
     raw_field *= 10;
 
-    /* rotate raw_field from sensor frame to body frame */
-    rotate_field(raw_field, _compass_instance);
-
-    /* publish raw_field (uncorrected point sample) for calibration use */
-    publish_raw_field(raw_field, _compass_instance);
-
-    /* correct raw_field for known errors */
-    correct_field(raw_field, _compass_instance);
-
     _last_read_ms = AP_HAL::millis();
-    
-    if (!_sem->take(HAL_SEMAPHORE_BLOCK_FOREVER)) {
-        return;
-    }
-    _mag_accum += raw_field;
-    _accum_count++;
-    if (_accum_count == 10) {
-        _mag_accum /= 2;
-        _accum_count = 5;
-    }
-    _sem->give();
 
+    accumulate_sample(raw_field, _compass_instance);
```

Comment:
Should be passing 10 as max_samples

Reply:
I didn't want to change the behavior of the driver... Too many variables to test at the same time.

Reply:
I take this back... I should indeed pass 10 here.